app/services/requests_total_items_service.rb

```diff
@@ -0,0 +1,41 @@
+class RequestsTotalItemsService
+  def initialize(requests:)
+    @requests = requests
+  end
+
+  def calculate
+    return unless requests
+
+    request_items_array = []
+    items_names = Item.where(id: request_items_ids).as_json(only: [:id, :name])
+
+    request_items.each do |items|
+      items.map do |json|
+        request_items_array << [item_name(items_names, json['item_id']), json['quantity']]
+      end
+    end
+
+    request_items_array.inject({}) do |item, (quantity, total)|
+      item[quantity] ||= 0
+      item[quantity] += total.to_i
+      item
+    end
+  end
+
+  private
+
+  attr_accessor :requests
+
+  def request_items_ids
+    request_items.map { |jitem| jitem.map { |item| item["item_id"] } }.flatten
```

Comment:
What do you think about `request_items.flat_map` instead of `.map.flatten`? It may seem a premature optimization but we actually have a slight performance improvement using `flat_map`. :)